### PR TITLE
Remove unused event variants and fields from Issue Events

### LINKED ISSUE #219: 'bot merge' errors
Here: https://github.com/paritytech/polkadot/pull/1659#issuecomment-690290296

### PR DESCRIPTION
**Pull request by joao-paulo-parity**
fixes #219 

Takes the suggestion from https://github.com/paritytech/parity-processbot/issues/219#issuecomment-793974787 to trim down to the used variants only. It's quite likely more code can be removed throughout the project, but that's part of another issue (#243).

Initially, the approach was parsing the field as `String` and then matching them through an interface (#245); that approach was removed in favor of another suggestion (https://github.com/paritytech/parity-processbot/pull/244#discussion_r591245983).

### CODE REVIEW

File: .rustfmt.toml [hunk 1]
```diff
@@ -1,2 +1,3 @@
 hard_tabs = true
```
[the remark this answers is not in the document]
Comment (ordian) [reply .rustfmt.toml:1]: https://github.com/paritytech/substrate/pull/2156 ;)

[the remark this answers is not in the document]
Comment (ordian) [reply .rustfmt.toml:1]: feel free to change that if you'll be maintaining the code

File: src/github.rs [hunk 2]
```diff
@@ -213,61 +214,19 @@ pub struct Contents {
...
-	Unassigned,
-	Unlabeled,
-	Unlocked,
-	UnmarkedAsDuplicate,
-	Unsubscribed,
-	UserBlocked,
+#[derive(
+	Debug, Clone, PartialEq, Serialize, Deserialize, strum_macros::ToString,
+)]
+#[allow(non_camel_case_types)]
```
Comment (ordian) [inline src/github.rs:220]: can use `#[strum(serialize_all = "snake_case")]` instead

File: src/github.rs [hunk 3]
```diff
@@ -213,61 +214,19 @@ pub struct Contents {
...
-	Subscribed,
-	Transferred,
-	Unassigned,
-	Unlabeled,
-	Unlocked,
-	UnmarkedAsDuplicate,
-	Unsubscribed,
-	UserBlocked,
+#[derive(
+	Debug, Clone, PartialEq, Serialize, Deserialize, strum_macros::ToString,
```
Comment (ordian) [inline src/github.rs:218]: only `ToString` is actually needed? serde is unused, but you could generate `ToString` with serde as well

Comment (joao-paulo-parity) [reply src/github.rs:217]: > only ToString is actually needed?

So far, yes

>  you could generate ToString with serde as well

Couldn't find a way to do that. Please post a coded suggestion.

Comment (ordian) [reply src/github.rs]: you could use `serde_json::to_string`, but I realized, you could also just implement a enum with unknown field:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=dbda67d6e5004041d932a0526764362a

Comment (joao-paulo-parity) [reply src/github.rs]: Suggestion taken. Thanks for the review!

**Review by haikoschol — APPROVED** [review]

**Pull request merged**